resources/scripts/mem.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+# Source:
+#   https://gist.github.com/bollig/56c277501505f5eaa5d185b86c2999b9
+# Usage:
+# 1. Set .5GB of Memory reserved for system processes
+#    ```
+#    sh Mem.sh 512  # 512 MB
+#    ```
+#  -or-
+# 2. Default (4GB reserved):
+#    ```
+#    sh Mem.sh
+#    ```
+# You'll need to include the following IAM permissions in your cluster's config:
+# ```
+# Iam:
+#   AdditionalIamPolicies:
+#       - Policy: arn:aws:iam::aws:policy/AmazonEC2ReadOnlyAccess
+# ```
+MEM=${1:-4096}
+FILELIST=$(ls /opt/slurm/etc/pcluster/slurm_parallelcluster*_partition.conf)
+
+for PARTFILE in $FILELIST; do
+        echo "Writiting to /tmp/$(basename $PARTFILE)"
+        while IFS= read -r line; do
+            if [[ $line == NodeName* ]]; then
+                    # NodeName=spot-dy-c52xlarge-[1-16] CPUs=4 State=CLOUD Feature=dynamic,c5.2xlarge,sp-small
+                    instance_type=$(echo "$line" | awk '{print $4}' | cut -f 2 -d',' )
+
+                    memory=$(AWS_DEFAULT_REGION=us-east-1 aws ec2 describe-instance-types --instance-type $instance_type --query "InstanceTypes[*].MemoryInfo[]" --output text)
+                    MemReserve=$MEM # MB
+                    RealMemory=`expr $memory - $MemReserve`
```

Comment:
What about the following:
```
                    RealMemory=$(python -c 'import sys; print(int(0.85 * int(sys.argv[1])))' $memory)
```

Reply:
That worked! When I ran it on a cluster with `t2.micro` (`1024 MB` Mem), I get `870 MB` set as RealMem:

```
[root@ip-172-31-27-232 ec2-user]# scontrol show node queue1-dy-compute-resource1-1
NodeName=queue1-dy-compute-resource1-1 Arch=x86_64 CoresPerSocket=1
   CPUAlloc=1 CPUTot=1 CPULoad=0.33
   AvailableFeatures=dynamic,t2.micro,compute-resource1
   ActiveFeatures=dynamic,t2.micro,compute-resource1
   Gres=(null)
   NodeAddr=172.31.17.170 NodeHostName=queue1-dy-compute-resource1-1 Version=21.08.6
   OS=Linux 4.14.262-200.489.amzn2.x86_64 #1 SMP Fri Feb 4 20:34:30 UTC 2022
   RealMemory=870 AllocMem=0 FreeMem=229 Sockets=1 Boards=1
   State=ALLOCATED+CLOUD ThreadsPerCore=1 TmpDisk=0 Weight=1 Owner=N/A MCS_label=N/A
   Partitions=queue1
   BootTime=2022-04-27T05:17:04 SlurmdStartTime=2022-04-27T05:19:30
   LastBusyTime=2022-04-27T05:19:30
   CfgTRES=cpu=1,mem=870M,billing=1
   AllocTRES=cpu=1
   CapWatts=n/a
   CurrentWatts=0 AveWatts=0
   ExtSensorsJoules=n/s ExtSensorsWatts=0 ExtSensorsTemp=n/s
```